### PR TITLE
Correctly convert sats to millisats in invoice creation

### PR DESCRIPTION
**Pull request by benthecarman**
When debugging I broke this in #76

### CODE REVIEW

**Comment by benthecarman** [comment]
Found some other bugs and threw the fixes in here

File: node-manager/src/event.rs [hunk 1]
```diff
@@ -271,14 +274,15 @@ impl LdkEventHandler for EventHandler {
...
+                match self.persister.read_payment_info(*payment_hash, false) {
                     Some(mut saved_payment_info) => {
                         saved_payment_info.status = HTLCStatus::Succeeded;
                         saved_payment_info.preimage = Some(payment_preimage.0);
+                        saved_payment_info.fee_paid_msat = *fee_paid_msat;
                         match self.persister.persist_payment_info(
                             *payment_hash,
                             saved_payment_info,
-                            true,
+                            false,
```
Comment (TonyGiorgio) [inline node-manager/src/event.rs:285]: i can't believe i screwed some of these up, i thought i triple checked them. nice catch.

File: node-manager/src/nodemanager.rs [hunk 2]
```diff
@@ -502,33 +508,45 @@ impl NodeManager {
...
+            Some(payment_info) => {
+                let mut mutiny_invoice = MutinyInvoice::from(invoice);
+
+                mutiny_invoice.is_send = true;
+                mutiny_invoice.paid = true;
+                mutiny_invoice.fees_paid = payment_info.fee_paid_msat;
+
+                Ok(mutiny_invoice)
+            }
+            None => Err(MutinyJsError::ReadError),
```
Comment (TonyGiorgio) [inline node-manager/src/nodemanager.rs:539]: Unfortunately this won't work. the invoice payer is async so it really only indicates that a payment is in progreess.

File: node-manager/src/nodemanager.rs [hunk 3]
```diff
@@ -542,27 +560,34 @@ impl NodeManager {
...
+                    bolt11: "".to_string(),
+                    description: None,
+                    payment_hash: payment_hash.0.to_hex(),
+                    preimage: Some(preimage.0.to_hex()),
+                    amount_sats: Some(amt_sats),
+                    expire: 0,
+                    paid: true,
+                    fees_paid: payment_info.fee_paid_msat,
+                    is_send: true,
+                };
```
Comment (TonyGiorgio) [inline node-manager/src/nodemanager.rs:585]: same with this

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]

File: node-manager/src/nodemanager.rs (hunk 2 above)
Comment (TonyGiorgio) [reply node-manager/src/nodemanager.rs:539]: In order to have a blocking call, we need a channel between event handler and this that blocks on waiting for the event handler to emit a payment success event.

Comment (benthecarman) [reply node-manager/src/nodemanager.rs]: reverted these changes as they'll be addresses in your PR anyways

**Pull request merged**